src/flags.jl

```diff
@@ -0,0 +1,226 @@
+"""
+function isproperpair(record::XAMRecord)::Bool
+    return flag(record) & PROPER_PAIR == PROPER_PAIR
+end
+
+"""
+    isunmapped(record::XAMRecord)::Bool
+
+Query whether the `record` is unmapped.
+"""
+function isunmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_UNMAP == FLAG_UNMAP
+end
+
+"""
+    ismapped(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped.
+"""
+function ismapped(record::XAMRecord)::Bool
+    # return flag(record) & FLAG_UNMAP == 0
+    return isfilled(record) && (flag(record) & FLAG_UNMAP == 0)
+end
+
+"""
+    ismateunmapped(record::XAMRecord)::Bool
+
+Query whether the `record`'s mate is unmapped.
+"""
+function ismateunmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_MUNMAP == FLAG_MUNMAP
+end
+
+"""
+    isnextmapped(record::XAMRecord)::Bool
+
+Test if the mate/next read of `record` is mapped.
+"""
+function isnextmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_MUNMAP == 0
+end
+
+"""
+    isreverse(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped to the reverse strand.
+"""
+function isreverse(record::XAMRecord)::Bool
+    return flag(record) & FLAG_REVERSE == FLAG_REVERSE
+end
+
+"""
+    isforward(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped to the forward strand.
+"""
+function isforward(record::XAMRecord)::Bool
+    return flag(record) & FLAG_REVERSE == 0
+end
+
+"""
+    ispositivestrand(record::XAMRecord)::Bool
+
+Query whether `record` is aligned to the positive strand.
+"""
+function ispositivestrand(record::XAMRecord)::Bool
+    return isforward(record)
+end
+
+"""
+    ispositivestrand(record::XAMRecord)::Bool
+
+Query whether `record` is aligned to the negative strand.
+"""
+function isnegativestrand(record::XAMRecord)::Bool
+    return isreverse(record)
+end
+
+"""
+    ismatereverse(record::XAMRecord)::Bool
+
+Query whether the `record`'s mate is mapped to the reverse strand.
+"""
+function ismatereverse(record::XAMRecord)::Bool
+    return flag(record) & FLAG_MREVERSE == FLAG_MREVERSE
+end
+
+"""
+    isread1(record::XAMRecord)::Bool
+
+Query whether the `record` is read1.
+"""
+function isread1(record::XAMRecord)::Bool
+    return flag(record) & FLAG_READ1 == FLAG_READ1
+end
+
+"""
+    isread2(record::XAMRecord)::Bool
+
+Query whether the `record` is read2.
+"""
+function isread2(record::XAMRecord)::Bool
+    return flag(record) & FLAG_READ2 == FLAG_READ2
+end
+
+"""
+    issecondaryalignment(record::XAMRecord)::Bool
+
+Query whether the `record` is a secondary alignment.
+"""
+function issecondaryalignment(record::XAMRecord)::Bool
+    return flag(record) & FLAG_SECONDARY == FLAG_SECONDARY
+end
+
+"""
+    isprimaryalignment(record::XAMRecord)::Bool
+
+Query whether the `record` is the primary alignment.
+"""
+function isprimaryalignment(record::XAMRecord)::Bool
+    return flag(record) & FLAG_SECONDARY == 0
```

Comment:
What is the difference between this and `isprimary`, which has a different definition? I would be clear in the documentation: What do you call a record which is neither secondary nor supplementary? What about one which is merely not supplementary?

Reply:
The `isprimaryalignment` method is wrong. A primary alignment is neither secondary nor supplementary, which is `0x100 | 0x800 = 0x900`, the primary line of the read.

Reply:
I've corrected this by merging `isprimary`, used internally, into `isprimaryalignment`.